hpcgap/lib/vecmat.gd

```diff
@@ -431,6 +431,34 @@ DeclareOperation( "ImmutableMatrix",[IsObject,IsMatrix]);
 
 #############################################################################
 ##
+#F  ImmutableVector( <field>, <vector>[, <change>] ) . convert into "best" representation
+##
+##  <#GAPDoc Label="ImmutableVector">
+##  <ManSection>
+##  <Oper Name="ImmutableVector" Arg='field, vector[, change]'/>
+##
+##  <Description>
+##  returns an immutable vector equal to <A>vector</A> which is in the optimal
+##  (concerning space and runtime) representation for vectors defined over
+##  <A>field</A>. This means that vectors obtained by several calls of
+##  <Ref Oper="ImmutableVector"/> for the same <A>field</A> are compatible
+##  for fast arithmetic without need for field conversion.
+##  <P/>
+##  The input vector <A>vector</A> might change the representation,
```

Comment:
I am not sure what you even mean with "is inconsistent with `Immutable`" -- so is `Print` ... :-). It is, however, consistent with many other GAP functions which silently change the rep of its inputs as it converts them into "better" formats  (e.g. for strings, blists, sets).

And this is not just the "long-term plan" -- if you look at the existing HPC-GAP implementation contained in this PR, it already today does not perform an in-place transformation.